### PR TITLE
server: try to make test TestTopSQLAgent stable

### LINKED ISSUE #25391: Unstable test `tidbTestTopSQLSuite.TestTopSQLAgent`
## Bug Report

Please answer these questions before submitting your issue. Thanks!
```
[2021-06-14T13:28:24.499Z] ----------------------------------------------------------------------
[2021-06-14T13:28:24.499Z] FAIL: tidb_test.go:1379: tidbTestTopSQLSuite.TestTopSQLAgent
[2021-06-14T13:28:24.499Z] 
[2021-06-14T13:28:24.499Z] tidb_test.go:1498:
[2021-06-14T13:28:24.499Z]     checkFn(0)
[2021-06-14T13:28:24.499Z] tidb_test.go:1423:
[2021-06-14T13:28:24.499Z]     c.Assert(len(records), Equals, n)
[2021-06-14T13:28:24.499Z] ... obtained int = 5
[2021-06-14T13:28:24.499Z] ... expected int = 0
```

### 1. Minimal reproduce step (Required)
in ci  https://ci.pingcap.net/blue/rest/organizations/jenkins/pipelines/tidb-unit-test-nightly/runs/655/nodes/127/steps/496/log/?start=0

### 4. What is your TiDB version? (Required)
master a45b3aaa2b1ac55bcce8d5a5d0018c9f56bc3f06

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve?

Try to fix #25391

### Check List 

Tests 

- No code

Side effects

- N/A

### Release note 

- No release note

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Review by xhebox — CHANGES_REQUESTED** [review]
LGTM CANCEL, since there is hold

**Comment by crazycs520** [comment]
@xhebox @breeswish PTAL

File: server/tidb_test.go [hunk 1]
```diff
@@ -1499,19 +1513,20 @@ func (ts *tidbTestTopSQLSuite) TestTopSQLAgent(c *C) {
...
 	cancel = runWorkload(11, 20)
+	// agent server shutdown
+	agentServer.Stop()
 	// agent server restart
 	agentServer, err = mockTopSQLReporter.StartMockAgentServer()
 	c.Assert(err, IsNil)
 	dbt.mustExec(fmt.Sprintf("set @@tidb_top_sql_agent_address='%v';", agentServer.Address()))
 	// check result
-	agentServer.WaitCollectCnt(1, time.Second*4)
+	agentServer.WaitCollectCnt(2, time.Second*8)
```
Comment (AilinKid) [inline server/tidb_test.go:1529]: once the exit with timeout here (possible), will the next `checkFn(5)` panic?

Comment (crazycs520) [reply server/tidb_test.go:1529]: If timeout, next `checkFn(5)` will not panic, but failed.

**Review by xhebox — APPROVED** [review]

**Review by breeswish — APPROVED** [review]

**Pull request merged**